keps/sig-cli/3805-ssa-default/README.md

```diff
@@ -0,0 +1,567 @@
+`--server-side=false` or `--force-conflict=true`.
+
+## Design Details
+
+As mentioned in the proposal, the new value of `auto` will detect if a
+resource has been client-side applied before and will continue to
+client-side apply these resources, while all other resources will be
+server-side applied. This means that new resources will be server-side
+applied by default.
+
+The `last-applied` annotation is used to detect previous client-side
+applied object, hence it is not inserted for server-side objects.
+
+`kubectl apply` flags that are specific to client-side apply will
+continue to work for client-side applied objects (prune, overwrite),
+while server-side apply flags will apply to existing server-side applied
+resources and new resources. Notably, the `--force-conflict` flag is
+somewhat incompatible with `--server-side=false` today, it will be
+possible to use the flag with `auto` (and we will entirely disallow it
+with `--server-side=false`).
+
+Some other commands might be impacted. `kubectl create` (and family)
+notably have a `--save-config` flag that create the last-applied
+annotation. While I don't know how many people actually use the flag,
+the idea of saving this as a config is confusing, since people don't
+actually have the file and so the situation doesn't really fit well the
+`apply` workflow. We suggest adding a warning when this flag is used, as
+well as updating its documentation to suggest not using it, and possibly
+deprecate it in the future.
+
+Because `kubectl diff` is supposed to map the behavior of `kubectl
+apply` as closely as possible, the change will also be done for that
+command.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
```

Comment:
I think these sections need to be filled out when targeting a release at all. Since the ApplySet KEP also touches apply it has some recently fetched numbers you can grab: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/3659-kubectl-apply-prune#test-plan